### ISSUE TITLE
SelectedTabIndex issue

### ISSUE THREAD

**Issue by mdziadowiec** [fluentribbon/Fluent.Ribbon#470]
**SelectedTabIndex doesn't work**

Hi,
I'm trying to use Ribbon Property SelectedTabIndex="{Binding SelectedTabIndex}" to activate tab from ViewModel but it doesn't do anything.
I particularly want to select a group tab active after it's set to visible with binding from ViewModel because it doesn't activate automaticly after getting visible:

    <fluent:RibbonContextualTabGroup Header="{Binding CurrentDzial.Nazwa}" 
                                                 Visibility="{Binding DzialContextToolsVisible, Converter={StaticResource BoolToVisibilityConverter}}"
                                                 x:Name="DzialGroup" Background="Brown" BorderBrush="Brown" /> 

I have made it work with code behind:

       private void Ribbon_OnIsVisibleChanged(object sender, DependencyPropertyChangedEventArgs e)
        {
            var item = sender as RibbonTabItem;

            if (item == null)  return;
           if (item.IsLoaded && !item.IsSelected && (bool) e.NewValue)
            {
                item.IsSelected = true;
            }
            else if (!item.IsLoaded && item.Visibility == Visibility.Visible)
            {
                item.IsSelected = false;
                item.Visibility = Visibility.Collapsed;
            }
        }
and it works fine but when i have changed base class of MainWindow from RibbonWindow to MahApps MetroWindow it stops working altogether: TabGroups don't show at all, RibbonContextualTabGroup Visibility Binding seems to stop working.
Am i doing something wrong or there is an issue that needs to be fixed?
Thanks for help.

---
### Environment

- Fluent.Ribbon __v5.0.2.46__
- Windows __10__ v1703
- .NET Framework __4.6.2__

**Comment by batzen** [comment]
Selecting an item/tab by index should work.
Could you provide a repro for that?

If `RibbonContextualTabGroup` etc. don't show up you forgot to add a `RibbonTitleBar` to your `MetroWindow`.
Have a look at https://github.com/fluentribbon/Fluent.Ribbon/blob/develop/Fluent.Ribbon.Showcase/MahMetroWindow.xaml and https://github.com/fluentribbon/Fluent.Ribbon/blob/develop/Fluent.Ribbon.Showcase/MahMetroWindow.xaml.cs to get to know how to add one.
The bridge between those two is `IRibbonWindow`.

**Comment by mdziadowiec** [comment]
Hi batzen, thanks for your response.
I have added RibbonTitleBar and the IRibbonWindow implementation per your suggestion and the groups have showed up when Visibility set to true in xaml, but the binding for Visibility still didn't work.
I had to specifically set datacontext to window.datacontext (see code below) which wasn't necesery when window was RibbonWindow not MetroWindow. (I'm using mvvmlight ViewModelLocator to set up datacontext of mainwindow.)

    Visibility="{Binding Path=DataContext.DzialContextToolsVisible, ElementName=window, Mode=OneWay, Converter={StaticResource BoolToVisibilityConverter}}"

Selecting tab by index still doesn't work, i don't know what can be wrong since i'm just binding the SelectedTabIndex property of the Ribbon to viewModel property...
Again i can use IsVisibleChanged event handler to select a tab which works fine so i'm satisfied.
Thank you.
